_inc/lib/debugger/class-jetpack-cxn-test-base.php

```diff
@@ -0,0 +1,295 @@
+				break;
+			case 'outbound_requests':
+				$resolution = __( 'Please ask your hosting provider to confirm your server can make outbound requests to jetpack.com.', 'jetpack' );
+				break;
+			case 'support':
+				$resolution = __( 'Please contact support.', 'jetpack' ); // @todo: Link to support.
+				break;
+		}
+
+		return array(
+			'name'       => $name,
+			'pass'       => false,
+			'message'    => $message,
+			'resolution' => $resolution,
+		);
+	}
+
+	/**
+	 * Provide WP_CLI friendly testing results.
+	 *
+	 * @param string $group Testing group whose results we are outputting. Default "default". Use "all" for all tests.
+	 */
+	public function output_results_for_cli( $group = 'default' ) {
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			if ( Jetpack::is_development_mode() ){
+				WP_CLI::line( __( 'Jetpack is in development mode. Please deactivate development mode for accurate results. ') );
+			}
+			WP_CLI::line( __( 'TEST RESULTS:', 'jetpack' ) );
+			foreach ( $this->raw_results( $group ) as $test ) {
+				if ( true === $test['pass'] ) {
+					WP_CLI::log( WP_CLI::colorize( '%gPassed:%n  ' . $test['name'] ) );
+				} elseif ( 'skipped' === $test['pass'] ) {
+					WP_CLI::log( WP_CLI::colorize( '%ySkipped:%n ' . $test['name'] ) );
+					if ( $test['message'] ) {
+						WP_CLI::log( '         ' . $test['message'] ); // Number of spaces to "tab indent" the reason.
+					}
+				} else { // Failed.
+					WP_CLI::log( WP_CLI::colorize( '%rFailed:%n  ' . $test['name'] ) );
+					WP_CLI::log( '         ' . $test['message'] ); // Number of spaces to "tab indent" the reason.
+				}
+			}
+		}
+	}
+
+	/**
+	 * Provide single WP Error instance of all failures.
+	 *
+	 * @param string $group Testing group whose failures we want converted. Default "default". Use "all" for all tests.
+	 *
+	 * @return WP_Error|false WP_Error with all failed tests or false if there were no failures.
+	 */
+	public function output_fails_as_wp_error( $group = 'default' ) {
+		if ( $this->pass( $group ) ) {
+			return false;
+		}
+		$fails = $this->list_fails( $group );
+		$error = false;
+
+		foreach ( $fails as $result ) {
+			$code    = 'failed_' . $result['name'];
+			$message = $result['message'];
+			$data    = array(
+				'resolution' => $result['resolution'],
+			);
+			if ( ! $error ) {
+				$error = new WP_Error( $code, $message, $data );
+			} else {
+				$error->add( $code, $message, $data );
+			}
+		}
+
+		return $error;
+	}
+
+	/**
+	 * Encrypt data for sending to WordPress.com.
+	 *
+	 * @param string $data Data to encrypt with the WP.com Public Key.
+	 *
+	 * @return false|array False if functionality not available. Array of encrypted data, encryption key.
+	 */
+	public function encrypt_string_for_wpcom( $data ) {
+		if ( ! function_exists( 'openssl_get_publickey' ) || ! function_exists( 'openssl_seal' ) ) {
+			return false;
+		}
+
+		$public_key = openssl_get_publickey( JETPACK__DEBUGGER_PUBLIC_KEY );
```

Comment:
As I understand it, this "should" be paired with a `openssl_free_key()`. I doubt it matters, though, since the PHP process ends shortly hereafter.

Reply:
I've found the documentation on openssl in php pretty lacking, so if you have any pointers, I'm all ears. Seems simple enough to free it. Added in 24d19da37